ipaserver/install/plugins/update_ra_cert_store.py

```diff
@@ -0,0 +1,115 @@
+from ipaplatform.constants import constants
+from ipaplatform.paths import paths
+from ipaplatform.tasks import tasks
+from ipapython import certdb
+
+register = Registry()
+
+
+@register()
+class update_ra_cert_store(Updater):
+    """
+    Moves the cert store from /etc/httpd/alias to /var/lib/ipa/radb
+    """
+
+    def execute(self, **options):
+        olddb = certdb.NSSDatabase(nssdir=paths.HTTPD_ALIAS_DIR)
+        if not olddb.has_nickname('ipaCert'):
+            # Nothign to do
+            return False, []
+
+        newdb = certdb.NSSDatabase(nssdir=paths.IPA_RADB_DIR)
+        if os.path.exists(paths.IPA_RADB_DIR):
+            if newdb.has_nickname:
+                self.log.warning(
+                    "An 'ipaCert' nickname exists in both the old {} and the "
+                    "new {} NSS Databases!".format(paths.HTTPD_ALIAS_DIR,
+                                                   paths.IPA_RADB_DIR))
+                return False, []
+        else:
+            # Get the owning user ids
+            pent = pwd.getpwnam(constants.IPAAPI_USER)
+
+            # create the initial directory
+            os.makedirs(paths.IPA_RADB_DIR, 0o751)
+            os.chown(paths.IPA_RADB_DIR, pent.pw_uid, pent.pw_gid)
+            tasks.restore_context(paths.IPA_RADB_DIR)
+
+            # Create the password file for this db
+            hex_str = binascii.hexlify(os.urandom(10))
+            f = os.open(paths.IPA_RADB_PWDFILE_TXT, os.O_CREAT | os.O_RDWR)
+            os.write(f, hex_str)
+            os.close(f)
+
+            # And create the DB
+            newdb.create_db(paths.IPA_RADB_PWDFILE_TXT)
+
+        # Import cert chain (ignore errors, as certs may already be imported)
+        certlist = olddb.list_certs()
+        certflags = {}
+        for name, flags in certlist:
+            certflags[name] = flags
+        for name in olddb.get_trust_chain('ipaCert'):
+            if name == 'ipaCert':
+                continue
+            try:
+                cert = olddb.get_cert(name, pem=True)
+                newdb.add_cert(cert, name, certflags[name], pem=True)
+            except Exception as e:  # pylint disable=broad-except
+                self.log.warning("Failed to import '{}' from trust "
+                                 "chain: {}".format(name, str(e)))
+
+        # As the last step export/import/delete the RA Cert
+        pw = binascii.hexlify(os.urandom(10))
+        p12file = os.path.join(paths.IPA_RADB_DIR, 'ipaCert.p12')
+        olddb.export_pkcs12('ipaCert', p12file, paths.ALIAS_PWDFILE_TXT, pw)
+        newdb.import_pkcs12(p12file, paths.IPA_RADB_PWDFILE_TXT, pw)
+
+        certmonger.stop_tracking(secdir=paths.HTTPD_ALIAS_DIR,
+                                 nickname='ipaCert')
+        certmonger.start_tracking(secdir=paths.IPA_RADB_DIR,
+                                  nickname='ipaCert',
+                                  password_file=paths.IPA_RADB_PWDFILE_TXT)
```

Comment:
No, this is about properly tracking the cert that has been moved from one place to another, certificate_reqnewal_upgrade() deals with renewing certs, but does not have conditional code to check if they moved, it would make things more complex and more obscure to do the re-tracking there instead of when and where the cert is being moved.